docker-plugin/src/main/java/com/nirima/jenkins/plugins/docker/builder/DockerBuilderControlOptionRun.java

```diff
@@ -87,10 +89,16 @@ public void execute(AbstractBuild<?, ?> build) throws DockerException, IOExcepti
 
         LOGGER.info("Pulling image " + xImage);
 
-        InputStream result = client.pullImageCmd(xImage).exec();
+        PullImageResultCallback resultCallback = new PullImageResultCallback() {
+            public void onNext(PullResponseItem item) {
+                if (item.getStatus() != null && item.getProgress() == null) {
```

Comment:
if you pull a layer there will be this nice `[====>            ]` progress indicator. So the status will be the same with an updated progress until the layer was pulled completely. I thought it would be enough to log every status once and not the progress.